src/imap.rs

```diff
@@ -1445,15 +1475,15 @@ impl Session {
                 .await
                 {
                     Ok(received_msg) => {
-                        if let Some(m) = received_msg {
-                            received_msgs.push(m);
-                        }
+                        received_msgs_channel
+                            .send((request_uid, received_msg))
+                            .await?;
                     }
                     Err(err) => {
                         warn!(context, "receive_imf error: {:#}.", err);
+                        received_msgs_channel.send((request_uid, None)).await?;
```

Comment:
It may be a temporary error, so at first glance it's better to continue processing messages, but not to send this particular UID. But as the message may be huge and we don't want to re-download it if the error is actually persistent, maybe at least add a comment?